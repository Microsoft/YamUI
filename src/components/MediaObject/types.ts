/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
export enum MediaObjectSize {
  XLARGE = 'xLarge',
  LARGE = 'large',
  MEDIUM = 'medium',
  SMALL = 'small',
  XSMALL = 'xSmall',
}
