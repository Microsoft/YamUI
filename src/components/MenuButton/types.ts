/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import { ContextualMenuItemType } from 'office-ui-fabric-react/lib/ContextualMenu';

export enum MenuItemType {
  Normal = ContextualMenuItemType.Normal,
  Header = ContextualMenuItemType.Header,
  Divider = ContextualMenuItemType.Divider,
}
