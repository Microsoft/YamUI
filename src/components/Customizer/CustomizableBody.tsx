/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import '../../yamui';
import * as React from 'react';
import { CustomizableComponentProps, defaultTheme, customizable } from './Customizer.types';
import { join } from '../../util/classNames';
import { NestableBaseComponentProps } from '../../util/BaseComponent/props';
import { getClassNames } from './Customizer.styles';

export class Body extends React.Component<CustomizableComponentProps & NestableBaseComponentProps> {
  public render() {
    const { children, className, theme = defaultTheme } = this.props;
    const classNames = getClassNames({ theme });
    const rootClassNames = join(['y-customizable-body', classNames.root, className]);

    return <div className={rootClassNames}>{children}</div>;
  }
}

@customizable('Body', ['theme'])
export default class CustomizableBody extends Body {}
