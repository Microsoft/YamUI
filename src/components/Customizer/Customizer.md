### Examples

Button with theme:

```js { "props": { "data-description": "basic" } }
const { createTheme, getTheme } = require('./');
const Button = require('../Button').default;

const grayTheme = createTheme({
  palette: {
    themeDark: '#333',
    neutralDark: '#666',
    themeLighter: '#999',
  },
  fonts: {
    smallPlus: {
      fontSize: '1.8rem',
      lineHeight: '2.4rem',
    },
  }
});

<div>
  <Button text="Normal" />{' '}
  <Customizer settings={{ theme: grayTheme }}>
    <Button text="Big Gray Theme" />
  </Customizer>
</div>;
```
