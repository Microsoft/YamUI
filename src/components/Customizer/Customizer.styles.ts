/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import { mergeStyleSets, ITheme } from 'office-ui-fabric-react/lib/Styling';
import { memoizeFunction } from 'office-ui-fabric-react/lib/Utilities';

export interface CustomizerClassNameProps {
  theme: ITheme;
}

export const getClassNames = memoizeFunction((styleProps: CustomizerClassNameProps) => {
  const { theme } = styleProps;
  const { fontSize, lineHeight, fontFamily, fontWeight } = theme.fonts.medium;

  return mergeStyleSets({
    root: {
      backgroundColor: theme.semanticColors.bodyBackground,
      color: theme.semanticColors.bodyText,
      fontSize,
      lineHeight,
      fontFamily,
      fontWeight,
    },
  });
});
