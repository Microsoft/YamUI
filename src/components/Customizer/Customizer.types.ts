/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import { ICustomizerProps, customizable } from 'office-ui-fabric-react/lib/Utilities';
import { createTheme, ITheme } from 'office-ui-fabric-react/lib/Styling';
import { theme as defaultTheme } from '../../util/theme';

export { createTheme, ITheme, ICustomizerProps, customizable, defaultTheme };

export interface CustomizableComponentProps {
  /**
   * Theme used to style the component. This will often be provided by `customizable`.
   */
  theme?: ITheme;
}
