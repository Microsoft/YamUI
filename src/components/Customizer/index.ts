/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
export { default, createTheme } from './Customizer';
export * from './Customizer';
