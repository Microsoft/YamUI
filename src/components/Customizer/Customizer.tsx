/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import '../../yamui';
import * as React from 'react';
import { NestableBaseComponentProps } from '../../util/BaseComponent/props';
import { Customizer as FabricCustomizer, ICustomizerProps } from 'office-ui-fabric-react/lib/Utilities';
import CustomizableBody from './CustomizableBody';

/**
 * Used to inject a custom theme object into customizable components, overriding the global theme.
 */
export default class Customizer extends React.Component<ICustomizerProps & NestableBaseComponentProps> {
  public render() {
    const { children, className, ...customizerProps } = this.props;

    return (
      <FabricCustomizer {...customizerProps}>
        <CustomizableBody {...{ children, className }} />
      </FabricCustomizer>
    );
  }
}
