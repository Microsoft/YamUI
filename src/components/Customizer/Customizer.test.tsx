/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import * as React from 'react';
import { shallow, mount, ShallowWrapper, ReactWrapper } from 'enzyme';
import Customizer from './Customizer';
import { ICustomizerProps, defaultTheme } from './Customizer.types';

describe('<Customizer />', () => {
  let component: ShallowWrapper<ICustomizerProps>;

  describe('with default options', () => {
    beforeEach(() => {
      component = shallow(<Customizer />);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with additional className', () => {
    beforeEach(() => {
      component = shallow(<Customizer className="TEST_CLASSNAME" />);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with children passed', () => {
    beforeEach(() => {
      component = shallow(<Customizer>CHILDREN</Customizer>);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with theme', () => {
    let mountedComponent: ReactWrapper;
    const theme = defaultTheme;

    beforeEach(() => {
      mountedComponent = mount(<Customizer settings={{ theme }} />);
    });

    it('body receives custom theme', () => {
      expect(mountedComponent.find('CustomizableBody').prop('theme')).toBe(theme);
    });
  });
});
