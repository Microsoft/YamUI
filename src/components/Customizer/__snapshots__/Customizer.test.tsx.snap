// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Customizer /> with additional className matches its snapshot 1`] = `
<Customizer>
  <CustomizedBody
    className="TEST_CLASSNAME"
  />
</Customizer>
`;

exports[`<Customizer /> with children passed matches its snapshot 1`] = `
<Customizer>
  <CustomizedBody>
    CHILDREN
  </CustomizedBody>
</Customizer>
`;

exports[`<Customizer /> with default options matches its snapshot 1`] = `
<Customizer>
  <CustomizedBody />
</Customizer>
`;
